crates/ide-completion/src/completions/extern_crate.rs

```diff
@@ -0,0 +1,96 @@
+//! Completion for extern crates
+
+use ide_db::{FxHashSet, SymbolKind};
+use syntax::{
+    ast::{self, NameRef},
+    AstNode,
+};
+
+use crate::{context::CompletionContext, CompletionItem, CompletionItemKind};
+
+use super::Completions;
+
+pub(crate) fn complete_extern_crate(
+    acc: &mut Completions,
+    ctx: &CompletionContext<'_>,
+    name_ref: Option<&NameRef>,
+) {
+    let imported_extern_crates: FxHashSet<String> = ctx
+        .token
+        .parent_ancestors()
+        .find_map(ast::SourceFile::cast)
+        .map(|src_file| {
+            src_file
+                .syntax()
+                .children()
+                .into_iter()
+                .filter_map(ast::ExternCrate::cast)
+                .filter_map(|node| node.name_ref())
+                .map(|name| name.to_string())
+                .collect()
+        })
+        .unwrap_or_default();
+
+    let current_txt =
+        name_ref.as_ref().map(|name_ref| name_ref.to_string()).unwrap_or(String::new());
```

Comment:
We don't need to filter the names ourselves, that's something the clients (text editors) are doing for us.